server/server.js

```diff
@@ -1,52 +1,50 @@
-import express from "express";
-import path from "path";
+import jwt from "jwt-simple";
+import Promise from "bluebird";
+import { notifHandler } from "hull/lib/utils";
+import Hull from "hull";
+
 import devMode from "./dev-mode";
 import SegmentHandler from "./handler";
 import handlers from "./events";
-import jwt from "jwt-simple";
+
 import analyticsClientFactory from "./analytics-client";
 import updateUser from "./update-user";
-import ejs from "ejs";
 
-module.exports = function server(options = {}) {
-  const { Hull, hostSecret, onMetric } = options;
-  const { BatchHandler, NotifHandler, Routes, Middleware: hullClient } = Hull;
-  const { Readme, Manifest } = Routes;
-  const app = express();
+
+module.exports = function Server(options = {}) {
+  const { app, connector, onMetric } = options;
 
   if (options.devMode) {
     app.use(devMode());
   }
 
-  app.engine("html", ejs.renderFile);
-  app.set("views", path.resolve(__dirname, "..", "views"));
-  app.use(express.static(path.resolve(__dirname, "..", "dist")));
-  app.use(express.static(path.resolve(__dirname, "..", "assets")));
-
-  app.get("/manifest.json", Manifest(__dirname));
-  app.get("/", Readme);
-  app.get("/readme", Readme);
-
-  app.get("/admin.html", hullClient({ hostSecret, fetchShip: false }), (req, res) => {
+  app.get("/admin.html", connector.clientMiddleware(), (req, res) => {
```

Comment:
the `clientMiddleware` is applied at the app level it should not be needed here.